### PR TITLE
refactor(hooks): refactor `useAttrs`

### PR DESCRIPTION
**Pull request by sxzz**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/hooks/use-attrs/index.ts [hunk 1]
```diff
@@ -9,29 +11,26 @@ interface Params {
...
+    warn(
+      'use-attrs',
+      'getCurrentInstance() returned null. useAttrs() must be called at the top of a setup function'
+    )
+    return computed(() => ({}))
+  }
+
+  return computed(() =>
+    Object.fromEntries(
+      Object.entries(instance.proxy?.$attrs).filter(
```
Comment (JeremyWuuuuu) [inline packages/hooks/use-attrs/index.ts:29]: I think if you don't use reactive to make this `$attrs` reactive, it should be a plain object with no reactivity cc @SimonaliaChen for comfirmation.

Comment (sxzz) [reply packages/hooks/use-attrs/index.ts:29]: `instance.$proxy` is reactived

File: packages/hooks/use-attrs/index.ts [hunk 2]
```diff
@@ -9,29 +11,26 @@ interface Params {
...
 
-export default (params: Params = {}) => {
+export default (params: Params = {}): ComputedRef<Record<string, unknown>> => {
   const { excludeListeners = false, excludeKeys = [] } = params
-  const instance = getCurrentInstance()
-  const attrs = shallowRef({})
   const allExcludeKeys = excludeKeys.concat(DEFAULT_EXCLUDE_KEYS)
 
-  // Since attrs are not reactive, make it reactive instead of doing in `onUpdated` hook for better performance
-  instance.attrs = reactive(instance.attrs)
```
Comment (sxzz) [inline packages/hooks/use-attrs/index.ts:19]: The original method will change internal variable, maybe breaking in the future.

**Review by jw-foss — APPROVED** [review]

File: packages/hooks/use-attrs/index.ts (hunk 2 above)
Comment (JeremyWuuuuu) [reply packages/hooks/use-attrs/index.ts:19]: Just confirmed. 👍

**Pull request merged**